front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -211,10 +213,18 @@ const SpaceTimeChartWrapper = ({
     allTrainsProjected
   );
 
-  const splitPoints = useMemo<SplitPoint[]>(() => {
-    const pathsIndex = keyBy(paths, ({ id }) => id);
+  const pathsIndex = useMemo(() => keyBy(paths, ({ id }) => id), [paths]);
 
-    return (
+  const totalOccurencesBySelectedPacedTrainId = useMemo(() => {
```

Comment:
What do you think of this name ?
```suggestion
  const selectedTimetableOccurrencesNb = useMemo(() => {
```